mls-rs/src/group/framing.rs

```diff
@@ -419,6 +419,23 @@ impl MlsMessage {
 
         kp.to_reference(cipher_suite).await.map(Some)
     }
+
+    /// If this is a plaintext proposal, return the proposal reference that can be matched e.g. with
+    /// [`StateUpdate::unused_proposals`](super::StateUpdate::unused_proposals).
+    #[cfg(feature = "by_ref_proposal")]
+    #[cfg_attr(not(mls_build_async), maybe_async::must_be_sync)]
+    pub async fn proposal_reference<C: CipherSuiteProvider>(
+        &self,
+        cipher_suite: &C,
+    ) -> Result<Vec<u8>, MlsError> {
+        matches!(&self.payload, MlsMessagePayload::Plain(ptx) if ptx.content.content_type() == ContentType::Proposal)
+            .then_some(())
+            .ok_or(MlsError::UnexpectedMessageType)?;
```

Comment:
Other methods return `None` or an empty `Vec` when the message has a different type. Should we be consistent here?

Reply:
I wanted to avoid `Result<Option<Vec<u8>>>` which is complicated. But I don't feel strongly...